src/foam/nanos/dig/DigWebAgent.java

```diff
@@ -489,6 +489,14 @@ protected synchronized FObject daoPut(DAO dao, FObject obj)
     throws Exception
   {
     FObject oldObj = dao.find(obj);
+
+    /**
+     * Only for the case of Simplified api user entity
+     * switching to auth.user from api.user to merge
+     */
+    if ( obj instanceof net.nanopay.api.User )
+      obj = ((net.nanopay.api.User) obj).getUser();
```

Comment:
you can't refer to nanopay within the foam2 library .. it is a circular dependency.